libs/angular-routing/src/lib/url-parser.ts

```diff
@@ -5,4 +5,24 @@ export class UrlParser {
   parse(url: string, base?: string | URL): URL {
     return new URL(url, base);
   }
+
+  joinUrls(currentUrl: string, url: string): string {
+    const currentUrlSegments = currentUrl
+      .split('#')[0] // remove hash
+      .split('?')[0] // remove query params
+      .split('/');
+    const urlSegments = url.split('/');
+
+    return urlSegments
+      .reduce((segments, segment) => {
```

Comment:
I wouldn't have considered using a reducer like this.

Makes for a really clean transform.

Reply:
Hopefully, it's also easy to read.